flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/LocalBufferPool.java

```diff
@@ -231,25 +230,39 @@ private BufferBuilder toBufferBuilder(MemorySegment memorySegment) {
 
 	@Nullable
 	private MemorySegment requestMemorySegment(boolean isBlocking) throws InterruptedException, IOException {
+		if (!isBlocking) {
+			return requestMemorySegment();
+		} else {
+			MemorySegment segment = null;
+			while (segment == null) {
+				if ((segment = requestMemorySegment()) == null) {
+					try {
+						// wait until available
+						isAvailable().get();
```

Comment:
I like this way actually. But we also need to remove invalid `availableMemorySegments.notify()` in `LocalBufferPool#recycle()` after removing `wait`.